src/ir/cost.h

```diff
@@ -744,6 +744,14 @@ struct CostAnalyzer : public OverriddenVisitor<CostAnalyzer, CostType> {
     // Inspired by indirect calls, but twice the cost.
     return 12 + visit(curr->cont);
   }
+  CostType visitSuspend(Suspend* curr) {
+    // Cheaper than resume, since payloads cannot be partially applied.
```

Comment:
This is definitely my most dreaded file now :sweat_smile: 

What I meant in the original comment here is the following: If you call `resume` on a continuation, it may have been produced with `cont.bind`, meaning that we have partially applied some arguments to the continuation, and `resume` only needs to provide the remaining ones. That means that we need to perform some runtime check that determines how to handle payloads in `resume`.
In contrast, `suspend` always sees all the payloads statically: When you call `suspend`, you provide all the payloads at once.

But there are more considerations when talking about the performance of `suspend` and `resume`, so mentioning only partial applications in my comment was a bit misleading: `resume` may have to check if you are initiating execution for the first time vs continuing after a suspension, `suspend` may need to forward to an outer handler, etc.

Given that `suspend` works in tandem with `resume` (i.e., whenever you call `suspend` you continue execution at another `resume`) and they both do stack switching, I'm now feeling more inclined to just give `suspend` the same cost as `resume` and delete my comment on `suspend`. Does that sound okay?

Reply:
Thanks, makes sense.

Yes, giving them the same cost without comment seems fine. Trying to be too precise in the reasoning here doesn't make sense given how imprecise all the numbers actually are :)

Reply:
Great, I've updated the function for `suspend` accordingly. I've also realized that I forgot to include the cost of the operands when calculating the cost of `resume`, so I fixed that in one go.